catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/app/SetPenColorBrickTest.java

```diff
@@ -64,6 +67,57 @@ public void setUp() {
 		baseActivityTestRule.launchActivity();
 	}
 
+	@Category({Cat.AppUi.class, Level.Smoke.class})
+	@Test
+	public void testPenColoraluesWithFormulaEditor() {
+		onBrickAtPosition(brickPosition).checkShowsText(R.string.brick_pen_color);
+		onBrickAtPosition(brickPosition).onFormulaTextField(R.id.brick_set_pen_color_action_red_edit_text)
+				.checkShowsNumber(0)
+				.perform(click());
+		onView(withText(R.string.brick_context_dialog_formula_edit_brick))
+				.perform(click());
+		onFormulaEditor()
+				.performEnterFormula("1+2")
```

Comment:
split this test into multiple smaller ones.
set up a brick field with the formula "1+2" before the activity is started, then do the operation to check whether formula editor is correctly opened or not.

same with the negative number, etc.

this would also be a candidate to do via robolectric, please have a look at BrickSingleFormulaFieldTest to get started on how to perform something like this.
we currently try to move as many formula editor tests over to the JVM as robolectric tests, since especially the typing operations and fragment loading is excessively slow on espresso.

Reply:
ok, I'm going to change it immediately. thanks for pointing that out.